python/cugraph/structure/graph_classes.py

```diff
@@ -0,0 +1,742 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from .graph_implementation import (simpleGraphImpl,
+                                   simpleDistributedGraphImpl,
+                                   npartiteGraphImpl)
+import cudf
+import warnings
+
+
+# TODO: Move to utilities
+def null_check(col):
+    if col.null_count != 0:
+        raise ValueError("Series contains NULL values")
+
+
+class Graph:
+    class Properties:
+        def __init__(self, directed):
+            self.directed = directed
+            self.weights = False
+
+    def __init__(self, m_graph=None, directed=False):
+        self._Impl = None
+        self.graph_properties = Graph.Properties(directed)
+        if m_graph is not None:
+            if m_graph.is_multigraph():
+                elist = m_graph.view_edge_list()
+                if m_graph.is_weighted():
+                    weights = "weights"
+                else:
+                    weights = None
+                self.from_cudf_edgelist(elist,
+                                        source="src",
+                                        destination="dst",
+                                        edge_attr=weights)
+            else:
+                msg = (
+                    "Graph can only be initialized using MultiGraph "
+                    "or MultiDiGraph"
+                )
+                raise Exception(msg)
+
+    def __getattr__(self, name):
+        if self._Impl is None:
+            raise AttributeError(name)
+        if hasattr(self._Impl, name):
+            return getattr(self._Impl, name)
+        elif hasattr(self._Impl.properties, name):
+            return getattr(self._Impl.properties, name)
+        else:
+            raise AttributeError(name)
+
+    def __dir__(self):
+        return dir(self._Impl)
+
+    def from_cudf_edgelist(
+        self,
+        input_df,
+        source="source",
+        destination="destination",
+        edge_attr=None,
+        renumber=True
+    ):
+        """
+        Initialize a graph from the edge list. It is an error to call this
+        method on an initialized Graph object. The passed input_df argument
```

Comment:
The wording here could be better.  the function adds edge data to the graph.  and user's can currently only call this function once - but that will be changing in a future release.  The fact that the call is used to initialize the class is secondary